pkg/nsx/services/staticroute/staticroute.go

```diff
@@ -104,24 +103,25 @@ func (service *StaticRouteService) patch(orgId string, projectId string, vpcId s
 	return nil
 }
 
-func (service *StaticRouteService) DeleteStaticRouteByPath(orgId string, projectId string, vpcId string, id string) error {
+func (service *StaticRouteService) DeleteStaticRoute(nsxStaticRoute *model.StaticRoutes) error {
 	staticRouteClient := service.NSXClient.StaticRouteClient
-	staticroute := service.StaticRouteStore.GetByKey(id)
-	if staticroute == nil {
-		return nil
+	vpcInfo, err := common.ParseVPCResourcePath(*nsxStaticRoute.Path)
```

Comment:
Better to check nil pointer before consuming a resource (`nsxStaticRoute`) if we dedicate on the current function `DeleteStaticRoute`. Not a strong option if we could the safeguard on it from the caller side.

Reply:
Currently we assume fields like Id and Path always exists in the NSX resource and not checking before dereferencing. Let's keep this as consistency. We can consider check them all in a future PR if necessary.